tests/utils/test_usage_stats.py

```diff
@@ -1,5 +1,52 @@
+from click.testing import CliRunner
+from mock import MagicMock, patch
+
 import bentoml
+from bentoml.cli import create_bentoml_cli
 from bentoml.utils.usage_stats import _get_bento_service_event_properties
+from bentoml.yatai.proto.deployment_pb2 import (
+    Deployment,
+    DeleteDeploymentResponse,
+)
+from bentoml.yatai.status import Status
+from bentoml.yatai.yatai_service import get_yatai_service
+
+
+MOCK_DEPLOYMENT_NAME = 'mock_name'
+MOCK_FAILED_DEPLOYMENT_NAME = 'mock-failed'
+MOCK_DEPLOYMENT_NAMESPACE = 'mock_namespace'
+
+
+def mock_track_func(event, properties):
+    return event, properties
+
+
+def mock_deployment_pb(name=MOCK_DEPLOYMENT_NAME):
+    mock_deployment = Deployment(name=name, namespace='mock-namespace')
+    mock_deployment.spec.azure_functions_operator_config.location = 'mock-location'
+    mock_deployment.spec.operator = 4
+    mock_deployment.created_at.GetCurrentTime()
+    return mock_deployment
+
+
+def mock_delete_deployment(deployment_pb):
```

Comment:
it's better to create test specific mocks instead of having the logic hidden here, e.g.:

```
def mock_get_operator_func():
    def func(yatai_service, deployment_pb):
        operator = MagicMock()
        operator.delete.side_effect = lambda deployment_pb: DeleteDeploymentResponse(status=Status.OK())
        return operator

    return func


def mock_aborted_get_operator_func():
    def func(yatai_service, deployment_pb):
        operator = MagicMock()
        operator.delete.side_effect = lambda deployment_pb: DeleteDeploymentResponse(status=Status.ABORTED())
        return operator

    return func
```